src/ua.c

```diff
@@ -1965,3 +2049,9 @@ int uag_set_extra_params(const char *eprm)
 
 	return 0;
 }
+
+
+void ua_set_custom_hdrs(struct ua *ua, struct list *custom_headers)
+{
+	ua->custom_hdrs = custom_headers;
+}
```

Comment:
you should use mem_deref and mem_ref here.

the data in struct ua should not point to objects from the application.
the application object might go out of scope, which will make the
`ua->custom_hdrs` pointer point to invalid data (and perhaps a crash).

Reply:
Sorry, I can't understand how exactly to use `mem_deref` and `mem_ref`.
Should this look something like the following?
```c++
mem_ref(custom_headers);
ua->custom_hdrs = custom_headers;
mem_deref(custom_headers);
```